app/views/organizations/adoptable_pets/index.html.erb

```diff
@@ -13,7 +13,39 @@
 <section class="pb-5" id="adoptable_pets">
   <div class="container">
     <div class="row">
-      <%= render partial:"shared/pet_card", collection: @pets, as: :pet %>
-    </div>
+      <% @pets.each do |pet| %>
+        <div class="col-lg-4 col-md-6 p-0">
+          <%= render CardComponent.new(
+            card_options: {class: "card card-hover m-4"},
+            image_options: {src: pet.images.first, url: adoptable_pet_path(pet)}
+          ) do |c| %>
+            <%= c.with_badge do %>
+              <%= tag.span("Adoption Pending", class: "badge bg-info") if pet.has_adoption_pending?%>
+              <%= tag.span("Adopted", class: "badge bg-warning") if pet.is_adopted?%>
+            <% end %>
+            <%= c.with_body do %>
+              <ul class="list-group list-group-flush">
+                <li class="list-group-item d-flex justify-content-between align-items-center flex-wrap">
+                  <%= link_to adoptable_pet_path(pet), class: 'text-decoration-none' do %>
+                    <h3 class="card-title text-secondary mb-0">
+                      <%= pet.name %>
+                    </h3>
+                  <% end %>
+                  <% if current_user&.adopter_foster_account %>
+                    <div class='text-end' id="like_button_pet_<%= pet.id %>">
+                      <%= render LikeButtonComponent.new(pet: pet, like: @likes&.find { |lp| lp.pet_id == pet.id }) %>
```

Comment:
this `find` is an insidious little n*m query.  You're looping through (on average) half the likes for every single pet.  Instead, when the likes get gathered, can do this:
```
@likes_by_id = @likes.index_by(&:pet_id)
```
and then you'll have a map of <pet_id => like> that can be used as `like: @likes_by_id[pet.id]`  , transforming these repeated scans into a direct lookup.